### PR TITLE
shinyapps/posit cloud build logs

### PR DESCRIPTION
**Pull request by mslynch**
## Intent

Resolves https://github.com/rstudio/rsconnect-python/issues/427.
Resolves https://github.com/rstudio/rsconnect-python/issues/432.

## Type of Change

- [ ] Bug Fix            
- [x] New Feature       
- [ ] Breaking Change   

## Approach

When a shinyapps.io task fails, we call the API to:
1. Get the child of the failed task whose action is `image-build`
2. Get and print the logs of that child task

Posit Cloud application builds don't have accessible logs yet (but will in the future). When a Posit Cloud task fails, we call the API to get the logs of the failed task, and if logs are returned, print them.

## Automated Tests

Added tests for `ShinyappsService.do_deploy` and `CloudService.do_deploy`.

## Directions for Reviewers

Add a dependency to `my-shiny-app/requirements.txt` that doesn't exist on PyPI, then run `rsconnect deploy shiny my-shiny-app` to deploy to shinyapps.io.

## Checklist

- [x] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [x] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+### Added
+- Failed deploys to shinyapps.io will now output build logs. Posit Cloud application deploys will also output build logs once supported server-side.
```
Comment (omar-rs) [inline CHANGELOG.md:10]: maybe reword to "once server-side is supported"?

File: tests/test_api.py [hunk 2]
```diff
@@ -356,45 +393,81 @@ def test_prepare_redeploy_static(self):
...
+        self.cloud_client.create_bundle.assert_called_with(10, "application/x-tar", bundle_size, bundle_hash)
 
         assert prepare_deploy_result.app_id == 1
         assert prepare_deploy_result.application_id == 10
         assert prepare_deploy_result.app_url == "https://posit.cloud/content/1"
         assert prepare_deploy_result.bundle_id == 100
         assert prepare_deploy_result.presigned_url == "https://presigned.url"
         assert prepare_deploy_result.presigned_checksum == "the_checksum"
+
+    # def do_deploy(self, bundle_id, app_id):
```
Comment (omar-rs) [inline tests/test_api.py:432]: remove

**Review by omar-rs — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (mslynch) [reply CHANGELOG.md:10]: I'm referring to connect-based outputs here, not server-side rendering.

Comment (omar-rs) [reply CHANGELOG.md:10]: Oh, I misunderstood the comment.  The words `server-side` are automatically associated to static outputs in my mind.

**Review by mmarchetti — APPROVED** [review]
